src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java

```diff
@@ -0,0 +1,201 @@
+        potionRecipes.put(Material.PUFFERFISH, PotionType.WATER_BREATHING);
+        potionRecipes.put(Material.GOLDEN_CARROT, PotionType.NIGHT_VISION);
+        potionRecipes.put(Material.TURTLE_HELMET, PotionType.TURTLE_MASTER);
+        potionRecipes.put(Material.PHANTOM_MEMBRANE, PotionType.SLOW_FALLING);
+
+        fermSpiderEye.put(PotionType.SPEED, PotionType.SLOWNESS);
+        fermSpiderEye.put(PotionType.JUMP, PotionType.SLOWNESS);
+        fermSpiderEye.put(PotionType.INSTANT_HEAL, PotionType.INSTANT_DAMAGE);
+        fermSpiderEye.put(PotionType.POISON, PotionType.INSTANT_DAMAGE);
+        fermSpiderEye.put(PotionType.NIGHT_VISION, PotionType.INVISIBILITY);
+    }
+
+    public AutoBrewer(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+    }
+
+    @Override
+    protected void tick(Block b) {
+        BlockMenu menu = BlockStorage.getInventory(b.getLocation());
+
+        if (isProcessing(b)) {
+            int timeleft = progress.get(b);
+
+            if (timeleft > 0) {
+                ChestMenuUtils.updateProgressbar(menu, 22, timeleft, processing.get(b).getTicks(), getProgressBar());
+
+                if (ChargableBlock.isChargable(b)) {
+                    if (ChargableBlock.getCharge(b) < getEnergyConsumption()) {
+                        return;
+                    }
+
+                    ChargableBlock.addCharge(b, -getEnergyConsumption());
+                    progress.put(b, timeleft - 1);
+                }
+                else {
+                    progress.put(b, timeleft - 1);
+                }
+            }
+            else {
+                menu.replaceExistingItem(22, new CustomItem(new ItemStack(Material.BLACK_STAINED_GLASS_PANE), " "));
+
+                for (ItemStack item : processing.get(b).getOutput()) {
+                    menu.pushItem(item, getOutputSlots());
+                }
+
+                progress.remove(b);
+                processing.remove(b);
+            }
+        }
+        else {
+            MachineRecipe recipe = findRecipe(menu);
+
+            if (recipe != null) {
+                if (!InvUtils.fitAll(menu.toInventory(), recipe.getOutput(), getOutputSlots())) {
+                    return;
+                }
+
+                for (int slot : getInputSlots()) {
+                    menu.consumeItem(slot);
+                }
+
+                processing.put(b, recipe);
+                progress.put(b, recipe.getTicks());
+            }
+        }
+    }
+
+    private MachineRecipe findRecipe(BlockMenu menu) {
+        ItemStack input1 = menu.getItemInSlot(getInputSlots()[0]);
+        ItemStack input2 = menu.getItemInSlot(getInputSlots()[1]);
+
+        if (input1 == null || input2 == null) return null;
+
+        if (input1.getType().name().endsWith("POTION") || input2.getType().name().endsWith("POTION")) {
+            boolean slot = input1.getType().name().endsWith("POTION");
```

Comment:
You should probably create an "isPotion(ItemStack)" method for this instead of doing this multiple times, also